module/VuFind/src/VuFind/Db/Entity/Search.php

```diff
@@ -295,14 +315,34 @@ public function setSaved(bool $saved): static
         return $this;
     }
 
+    /**
+     * Post-load normalization (deserialization).
+     *
+     * @ORM\PostLoad
+     */
+    public function postLoadNormalize(): void
+    {
+        // Only deserialize if searchObject is not null and not already deserialized
+        if ($this->searchObject && !is_object($this->searchObject)) {
+            // If it's a resource (stream), convert it to a string first
+            if (is_resource($this->searchObject)) {
+                $this->searchObject = stream_get_contents($this->searchObject);
+            }
+            $unserialized = @unserialize($this->searchObject);
+            if ($unserialized && is_object($unserialized)) {
+                $this->searchObject = $unserialized; 
```

Comment:
Shouldn't this use deserializedSearchObject?

```suggestion
                $this->deserializedSearchObject = $unserialized; 
```

...and maybe we need an else clause to nullify the property if the unserialization fails.